calcom/cal.com
# chore: Refactor Large UI components.

## What does this PR do?

This progressively breaks down large components into smaller one and aims at making similar future refactoring easier.

- Refactored `BookListItem` and broke it down into a component responsible for all the actions. Also stripped away the standalone components found in the same file.
- Deduplicate code in `customer.ts` that handles stripe payment.


Fixes #12600



## Requirement/Documentation



- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change




## How should this be tested?



No behavioral change is expected from this PR, so the existing test cases are expected to pass.

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks


## Checklist



- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
